cli/cli_app.go

```diff
@@ -346,6 +346,18 @@ func RunTerragrunt(terragruntOptions *options.TerragruntOptions) error {
 		return err
 	}
 
+	if terragruntOptions.IamRole == "" {
+		terragruntFlags, err := config.PartialParseConfigFile(terragruntOptions.TerragruntConfigPath, terragruntOptions, nil, []config.PartialDecodeSectionType{config.TerragruntFlags})
```

Comment:
I'm having trouble thinking through the implications of doing a partial parsing here... E.g.,

1. What if the IAM role ARN to assume is read from a `dependency` block. Will that work?
1. What about an IAM role ARN read via `read_terragrunt_config`?
1. What about an `iam_role` param in an `include` block? Will that work?

@yorinasub17 Would appreciate another set of eyes on this.

Reply:
Thanks for your feedback @brikis98 , really appreciate it 👍🏼 , I was trying to get something for the regression tests on this, but seems there is no test for the `iam_role` functionality, according to  this @yorinasub17 PR [here](https://github.com/gruntwork-io/terragrunt/pull/1315) 
> NOTE: I didn't add tests because it doesn't look like we have tests for this functionality, but we should figure out a way to have a test. In the meantime, I am going to cut an alpha testing release so that the reporter can test this.

The idea was to assume the aws iam role as soon as possible and affect everything, so the AWS credentials will be available for other subcommands for instance `run_cmd`, I don't have the broad `terragrunt` picture and flow as clear as you may have guys , so any clue / feedback will be much apreaciated 

more info here https://github.com/gruntwork-io/terragrunt/issues/1636

Reply:
Hi @yorinasub17 , sorry to bother you, but I was wondering if you could give me some hints/clues,  about how could I implement regression tests for this scenario,  I have seen the `integration_test.go` file but could not come up with a good approach for that. 

Any idea will be much appreciated.

thanks for your time 👍🏼

Reply:
Hi @yorinasub17 , no problem, thanks so much for your time and ideas!, I have added the regression tests aligned with your recommendations, could you please have a look 👍🏼  

```bash
go test  -v -run ^TestTerragruntIamRoleProperty$
```

not sure though about case 1:

> What if the IAM role ARN to assume is read from a dependency block. Will that work?

any idea about how to tackle that one?